### PR TITLE
Feature/sorting by date method

### LINKED ISSUE #13: Sorting feature
As a user, I should be able to sort events by date
- [ ] From latest to earliest
- [ ] From earliest to latest

### PR DESCRIPTION
**Pull request by okalimbet**
#### What's this PR do?
Add sort method to sort postings by date
Fix state in App component based new changes.

#### What does it fix?
N/A

#### Where should the reviewer start?
App - index.js: line 14: 
* add isSorted state, 
* new `sortPostingsByDate` method, 
* new prop is passed to Postings component

Navigation - index.js: line 2: 
* new image for a glyph-icon was added, 
* add new button for sort, 
* new `sortByDate` function was added.

Postings - index.js: line 6: 
* a newly created sort function was passed from App component 

#### How should this be manually tested?
by running` npm start`

#### Any background context you want to provide?
Sort doesnt have an appropriate styling yet. Sort, filter, and search need styling fixed.

#### Screenshots (if appropriate):
Screen Shot 2021-01-07 at 5 06 24 PM

### CODE REVIEW

File: src/Components/App/index.js [hunk 1]
```diff
@@ -31,6 +32,18 @@ const App = () => {
     setQueriedPostings(queriedPostings);
   };
 
+  const sortPostingsByDate = () => {
+    if(!isSorted) {
+      postings.sort((a, b) => {
+        return (a.date > b.date) ? -1 : 1;
+      })
+    } else {
+      postings.reverse();
```
Comment (yiranli624) [inline src/Components/App/index.js:41]: Nice use of reverse!

File: src/Components/Navigation/index.js [hunk 2]
```diff
@@ -9,10 +13,24 @@ const Navigation = ({ searchByKeyWord, categories, filterByCategory }) => {
...
     <section>
       <input type="text" value={keyWord} placeholder="i.e Boulder..." onChange={(e) => setKeyWord(e.target.value)}></input>
       <input type="submit" value="search" onClick={(searchPostings)}></input>
+
+      <div class="btn-group sort-button-group pull-right"> 
+        <button 
+          onClick={sortPostings}
+          className ="button-sort"
+          value="sort"
+          type="submit">Sort<span aria-hidden="true" class="glyphicon"><img src={sortGlyphicon}/></span>
```
Comment (yiranli624) [inline src/Components/Navigation/index.js:31]: Love the big icon lol

**Review by yiranli624 — APPROVED** [review]
Great work with the sorting feature!

File: src/Components/Navigation/index.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 import React, { useState } from "react";
+import sortGlyphicon from "../../Assets/sort-button.png";
 
-const Navigation = ({ searchByKeyWord, categories, filterByCategory }) => {
+
```
Comment (coopterrones) [inline src/Components/Navigation/index.js:4]: what is this blank space...

Comment (coopterrones) [reply src/Components/Navigation/index.js:4]: ahh nvm

File: src/Components/Navigation/index.js [hunk 4]
```diff
@@ -9,10 +13,24 @@ const Navigation = ({ searchByKeyWord, categories, filterByCategory }) => {
     setKeyWord('');
   }
 
+  const sortPostings = (e) => {
```
Comment (coopterrones) [inline src/Components/Navigation/index.js:16]: sort looks great

File: src/Components/App/index.js [hunk 5]
```diff
@@ -31,6 +32,18 @@ const App = () => {
     setQueriedPostings(queriedPostings);
   };
 
+  const sortPostingsByDate = () => {
```
Comment (coopterrones) [inline src/Components/App/index.js:35]: sort looks great!

Comment (coopterrones) [reply src/Components/App/index.js:35]: We DEFINITELY need some re-styling.. we got dis

**Pull request merged**